testing/stubs.go

```diff
@@ -141,6 +141,115 @@ func TestFailOnceActivityInterceptor() activity.ActivityInterceptor {
 	}
 }
 
+func TestThrotteSignalsOnceInterceptor() fsm.DecisionInterceptor {
```

Comment:
it would help to have a comment here that explains the execution flow. i think i figured it out, but it's pretty convoluted because the after hook happens before the before hook. thinking just simple enumeration of the steps as a comment